addons/binding/org.openhab.binding.xiaomivacuum/ESH-INF/thing/vacuumThing.xml

```diff
@@ -0,0 +1,225 @@
+		</channels>
+	</channel-group-type>
+	<channel-group-type id="dnd">
+		<label>Do not Disturb</label>
+		<channels>
+			<channel id="dnd_function" typeId="dnd_function" />
+			<channel id="dnd_start" typeId="dnd_start" />
+			<channel id="dnd_end" typeId="dnd_end" />
+		</channels>
+	</channel-group-type>
+	<channel-group-type id="history">
+		<label>History</label>
+		<channels>
+			<channel id="total_clean_area" typeId="total_clean_area" />
+			<channel id="total_clean_time" typeId="total_clean_time" />
+			<channel id="total_clean_count" typeId="total_clean_count" />
+		</channels>
+	</channel-group-type>
+
+	<!-- Status channels -->
+	<channel-type id="clean_area">
+		<item-type>Number</item-type>
+		<label>Cleaning Area</label>
+		<state pattern="%.2d M2" readOnly="true" />
+	</channel-type>
+	<channel-type id="clean_time">
+		<item-type>Number</item-type>
+		<label>Cleaning Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="dnd_enabled">
+		<item-type>Switch</item-type>
+		<label>Do not Disturb</label>
+	</channel-type>
+	<channel-type id="error_code">
+		<item-type>String</item-type>
+		<label>Error</label>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="fan_power">
+		<item-type>Number</item-type>
+		<label>Fan Power</label>
+		<state pattern="%.0d %%" readOnly="true" />
+	</channel-type>
+	<channel-type id="in_cleaning">
+		<item-type>Number</item-type>
+		<label>in_cleaning</label>
+		<state pattern="%.0d" readOnly="true" />
+	</channel-type>
+	<channel-type id="map_present" advanced="true">
+		<item-type>Number</item-type>
+		<label>map_present</label>
+		<category>Energy</category>
+		<state pattern="%.0d" readOnly="true" />
+	</channel-type>
+	<channel-type id="msg_seq" advanced="true">
+		<item-type>Number</item-type>
+		<label>msg_seq</label>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="msg_ver" advanced="true">
+		<item-type>Number</item-type>
+		<label>msg_ver</label>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="state">
+		<item-type>String</item-type>
+		<label>state</label>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Consumables channels -->
+	<channel-type id="main_brush_time">
+		<item-type>Number</item-type>
+		<label>Main Brush Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="side_brush_time">
+		<item-type>Number</item-type>
+		<label>Side Brush Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="filter_time">
+		<item-type>Number</item-type>
+		<label>Filter Usage Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+	<channel-type id="sensor_dirt_time">
+		<item-type>Number</item-type>
+		<label>Sensor Dirt Time</label>
+		<state pattern="%.2d min" readOnly="true" />
+	</channel-type>
+
+	<!-- actions -->
+	<channel-type id="vacuum">
```

Comment:
Yes, I like the idea very much.
I think the current command channel will still be something separate, as that allows for arbitrary commands (not implemented in the binding) to be executed and fed back.
e.g. I plan to use that for a nice custom widget in habpanel to show the history data